### PR TITLE
Add option for providing a different SASS implementation

### LINKED ISSUE #137: Allow configuring sass implementation.
Currently, `true` uses `node-sass` by default, but I use `dart-sass` in my project. It is currently not possible to configure the sass implementation, so to use `true` I have to install `node-sass` in addition to `dart-sass`.

### PR DESCRIPTION
**Pull request by rhumaric**
Fixes #137 

Hadd the same issue as in #137 so I've added an option on this branch to pass a different SASS implementation.

I've added a run of the test against the `sass` implementation, which led to discovering a [couple of different behaviours](https://github.com/sass/dart-sass/#behavioral-differences-from-ruby-sass) for which I had to modify the tests a bit.

### CODE REVIEW

File: test/scss/test.scss [hunk 1]
```diff
@@ -6,6 +6,10 @@
 // Error Override
 $_true-error-output-override: true;
 
+// sass behaves slightly differently than node-sass
+// This flag will help adjust tests when needed
+$_dart-sass: false !default;
```
Comment (mirisuzanne) [inline test/scss/test.scss]: I'd rather we find a way to make things work consistently on our end, and let users handle different implementations. I'd have to dig deeper to see how that might play out for these cases – and I'm open to ideas. But I'd rather not be testing different Sass implementations here.

Comment (rhumaric) [reply test/scss/test.scss]: Fair point. I did it like that to check that everything would run fine with `sass`, which was my main use case. Understandable that it's not in the scope of `sass-true` to check every SASS implementation. I'll update the tests and use a spy on a JS function to verify that `runSass` does run the SASS implementation it is provided.

Actually, in terms of JS API, I was thinking it's not that great to merge that new option in the middle of the options passed to SASS. However, adding a 4th parameter doesn't sound that amazing either. Wondering if you'd be open to a breaking change and have `runSass` called like this:

```js
sassTrue.runSass({filename: 'path/to/some.scss'}, {describe: describe, it: it, sassEngine: require('sass')}
```

Newer versions of node would even allow this more succinct version, as far as I understand (to be confirmed, though):

```js
sassTrue.runSass({filename: 'path/to/some.scss'},{describe,it, sassEngine: require('sass')}
```

**Comment by jackmcpickle** [comment]
I'm keen for this too. Let me know if I can help. Dart sass has deprecated !global now so keen to either allow switch or pass each engine in so I can run tests on both.

File: README.md [hunk 2]
```diff
@@ -156,17 +156,28 @@ when upgrading from an older version of True.
    npm install sass-true
    ```
 
-2. [Optional] Install `node-sass` (required as a `peerDependency`),
-   if it's not already installed in your project:
+2. [Optional] Install the SASS implementation matching your project (if not already installed)
```
Comment (mirisuzanne) [inline README.md]: SASS => Sass (it's not an acronym)

**Comment by rhumaric** [comment]
I've updated the tests to remove a second run of the tests with the `sass` package. Tests should be much cleaner now.

I've also had to bump the `node-sass` package for Travis build to run OK with Node 12. Looks like [the minimum version for this version of node is 4.12](https://github.com/sass/node-sass).

I haven't introduced the change of API I described. I think it could be done in a separate pull request once this one is merged so things are kept tidy.

Let me know if there's any other changes needed (maybe a quick rebase of the branch to have a cleaner history).

File: README.md (hunk 2 above)
Comment (rhumaric) [reply README.md]: Indeed. Updated that quickly.

File: README.md [hunk 3]
```diff
@@ -175,6 +186,19 @@ when upgrading from an older version of True.
...
+   **For `sass`/`dart-sass`**:
+
+   ```js
+   var path = require('path');
+   var sassTrue = require('sass-true');
+
+   var sassFile = path.join(__dirname, 'test.scss');
+   sassTrue.runSass({
+     file: sassFile,
+     sass = require('sass')
```
[the remark this answers is not in the document]
Comment (rhumaric) [reply README.md]: Oh, I hadn't noticed that, good spot. I'm surprised it even works, the intention was definitely `sass: require('sass')`. I'll fix that quickly.

[the remark this answers is not in the document]
Comment (rhumaric) [reply README.md]: Ah, it's in the README, so probably doesn't work at all.

[the remark this answers is not in the document]
Comment (rhumaric) [reply README.md]: Updated :)

**Comment by jgerigmeyer** [comment]
Thanks!

**Pull request merged**